### DISCUSSION TITLE
User Model & Relational Models

### DISCUSSION THREAD

**Discussion by malsgit** [nWidart/laravel-modules#1305] [Q&A]
Hello,

Just getting started with Laravel & Modules and want to make the Module so I can use in other projects, ordinarily with the User Model I would, put relational model stuff like below in app/Models/Users.php
```
public function posts()
    {
        return $this->hasMany(Post::class);
    }
```

How would I achieve the same in a Module, so all the logic for that module is contained in that module related to the User Specifically so I don't have to reference it in app/Models/User.php making it easier to use modules in other projects.

also how would I do the same for other relations from other modules/models?

Regards

M

**Comment by Chomiciak** [comment]
In it's essence, laravel's relation definition should return a collection. You can either load that collection based on the contents of a module, or make it another way, including returning just an empty one
